### PR TITLE
Add a lot more data to info.json

### PR DESCRIPTION
**Pull request by skullydazed**
## Description

This adds a lot of new information to `info.json`. Besides adding more keyboard metadata we have also added tags! Just add a new top-level key to info.json containing a list of your tags. Example:

```json
{
    "tags": ["tkl", "split"]
}
```

This also changes the output of `qmk info -f text` to use dot-notation. This will be handy in the future as we add the ability to query based on JSON information.

<details>
  <summary>Example qmk info -f text output</summary>
  
  ```
    skully@zayante:~/qmk_firmware:0$ qmk info -kb clueboard/card -f text
    bootloader: atmel-dfu
    debounce: 20
    diode_direction: ROW2COL
    features.audio: True
    features.backlight: True
    features.bluetooth: False
    features.bootmagic: False
    features.command: True
    features.console: True
    features.extrakey: True
    features.lto: True
    features.midi: False
    features.mousekey: True
    features.nkro: False
    features.rgblight: True
    features.unicode: False
    height: 8
    keyboard_folder: clueboard/card
    keyboard_name: Cluecard
    layout_aliases.LAYOUT: LAYOUT_all
    layouts: LAYOUT_all
    maintainer: skullydazed
    manufacturer: Clueboard
    matrix_pins.cols: F1, F6, F7
    matrix_pins.rows: B4, F0, F4, F5
    platform: unknown
    processor: atmega32u4
    processor_type: avr
    protocol: LUFA
    rgblight.brightness_steps: 17
    rgblight.hue_steps: 10
    rgblight.led_count: 4
    rgblight.pin: E6
    rgblight.saturation_steps: 17
    split.transport.protocol: serial
    usb.device_ver: 0x0001
    usb.pid: 0x2330
    usb.vid: 0xC1ED
    width: 10
  ```
</details>

## Types of Changes

- [x] Core
- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by noroadsleft — COMMENTED** [review]
`qmk info -h` seems to imply `qmk info -f` defaults to `friendly`, but this doesn't work in practice:

```
 noroadsleft  ~/qmk_firmware/keyboards/clueboard/66/rev4  (pr/13366)
$ qmk info -f
usage: qmk.exe info [-h] [-r] [--ascii] [-f FORMAT] [-m] [-l] [-km KEYMAP] [-kb KEYBOARD]
qmk.exe info: error: argument -f/--format: expected one argument

 noroadsleft  ~/qmk_firmware/keyboards/clueboard/66/rev4  (pr/13366)
$ qmk info -h
usage: qmk.exe info [-h] [-r] [--ascii] [-f FORMAT] [-m] [-l] [-km KEYMAP] [-kb KEYBOARD]

optional arguments:
  -h, --help            show this help message and exit
  -r, --rules-mk        Render the parsed values of the keyboard's rules.mk file.
  --ascii               Render layout box drawings in ASCII only.
  -f FORMAT, --format FORMAT
                        Format to display the data in (friendly, text, json) (Default: friendly).
  -m, --matrix          Render the layouts with matrix information.
  -l, --layouts         Render the layouts.
  -km KEYMAP, --keymap KEYMAP
                        Show the layers for a JSON keymap too.
  -kb KEYBOARD, --keyboard KEYBOARD
                        Keyboard to show info for.
```

**Comment by skullydazed** [comment]
Apologies for missing this last week.

> `qmk info -h` seems to imply `qmk info -f` defaults to `friendly`, but this doesn't work in practice:

You have to supply an argument to `-f`. If you don't supply `-f` at all it defaults to the friendly output.

> Specifying `friendly` also doesn't seem to do anything:

That's working as designed. If you were to override the default output, EG `qmk config info.format=text`, then using `qmk info -f friendly` would look different.

**Comment by skullydazed** [comment]
Tests are passing now and it can merge clean.

File: data/schemas/keyboard.jsonschema [hunk 1]
```diff
@@ -128,13 +235,118 @@
...
+                    "max": 5,
+                    "multipleOf": 1
+                },
+                "transport": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                        "protocol": {
+                            "type": "string",
+                            "enum": ["custom", "i2c", "serial"]
```
Comment (tzarc) [inline data/schemas/keyboard.jsonschema:283]: Most of the time people will want to use `serial_usart` for split on ARM.
```suggestion
                            "enum": ["custom", "i2c", "serial", "serial_usart"]
```

The `serial_usart` driver has its own set of configurations:
* `#define SERIAL_USART_FULL_DUPLEX` (just `#ifdef`, no `TRUE`, default is undef'd)
* `#define SERIAL_USART_PIN_SWAP` (just `#ifdef`, no `TRUE`, default is undef'd)
* `#define SERIAL_USART_TX_PIN B6` - matches mcu pin
* `#define SERIAL_USART_RX_PIN B7` - matches mcu pin
* `#define SERIAL_USART_TX_PAL_MODE 7` - uint8
* `#define SERIAL_USART_RX_PAL_MODE 7` - uint8
* `#define SERIAL_USART_TIMEOUT 100` - uint8
* `#define SERIAL_USART_DRIVER SD1` - name of the serial driver, probably safe to limit between `SD1` and `SD9` -- unsure how many the biggest chips have but this should be safe for a first pass.

File: data/mappings/info_config.json [hunk 2]
```diff
@@ -30,17 +55,40 @@
...
     "PRODUCT_ID": {"info_key": "usb.pid", "value_type": "hex"},
     "VENDOR_ID": {"info_key": "usb.vid", "value_type": "hex"},
     "QMK_ESC_OUTPUT": {"info_key": "qmk_lufa_bootloader.esc_output"},
     "QMK_ESC_INPUT": {"info_key": "qmk_lufa_bootloader.esc_input"},
+    "QMK_KEYS_PER_SCAN": {"info_key": "qmk.keys_per_scan", "value_type": "int"},
     "QMK_LED": {"info_key": "qmk_lufa_bootloader.led"},
-    "QMK_SPEAKER": {"info_key": "qmk_lufa_bootloader.speaker"}
+    "QMK_SPEAKER": {"info_key": "qmk_lufa_bootloader.speaker"},
+    "SPLIT_MODS_ENABLE": {"info_key": "split.transport.mods", "value_type": "bool"},
+    "SPLIT_TRANSPORT_MIRROR": {"info_key": "split.transport.mirror", "value_type": "bool"},
```
Comment (tzarc) [inline data/mappings/info_config.json:78]: I think this PR was raised before #11930 was merged -- all of these are new split-related defines.

I'd also advocate for some more relevant naming, seeing as we're introducing these:
```suggestion
    "SPLIT_MODS_ENABLE": {"info_key": "split.transport.sync_modifiers", "value_type": "bool"},
    "SPLIT_TRANSPORT_MIRROR": {"info_key": "split.transport.sync_matrix_state", "value_type": "bool"},
```

We'll also need these `bool`'s:
* `SPLIT_LAYER_STATE_ENABLE` - `split.transport.sync_layer_state`?
* `SPLIT_LED_STATE_ENABLE` - `split.transport.sync_led_state`?
* `SPLIT_WPM_ENABLE` - `split.transport.sync_wpm`?
* `SPLIT_OLED_ENABLE` - `split.transport.sync_oled`?
* `SPLIT_ST7565_ENABLE` - `split.transport.sync_st7565`?

And an unsigned int:
* `FORCED_SYNC_THROTTLE_MS` - `split.transport.sync_throttle_ms`?

**Review by tzarc — CHANGES_REQUESTED** [review]

File: data/schemas/keyboard.jsonschema (hunk 1 above)
Comment (skullydazed) [reply data/schemas/keyboard.jsonschema:283]: That option wasn't documented on <https://docs.qmk.fm/#/config_options>. :/

How critical is it to add these other options? For this PR I was trying to limit it to everything found in `docs/config_options.md`.

File: data/mappings/info_config.json (hunk 2 above)
Comment (skullydazed) [reply data/mappings/info_config.json]: Are these critical to get in now, or can we add them later?

**Comment by skullydazed** [comment]
I've rebased against develop and fixed a couple new errors that cropped up.

File: data/schemas/keyboard.jsonschema (hunk 1 above)
Comment (tzarc) [reply data/schemas/keyboard.jsonschema]: They are, on `develop`, using `qmk docs`:
image

**Review by tzarc — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**